.github/workflows/e2e-tests-main-devnet.yml

```diff
@@ -532,6 +532,148 @@ jobs:
+    needs: [ check-runtime-change ]
+    if: ${{ needs.check-runtime-change.outputs.runtime-updated != 0 }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.base_ref }}
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+
+      - name: Install WASM target
+        run: rustup target add wasm32-unknown-unknown
+
+      - name: Restore cache
+        uses: ./.github/actions/restore-cache
+        with:
+          target-key: try-runtime
+          cargo-key: old
+
+      - name: Build old binary
+        run: cargo build --release -p aleph-node --features "short_session"
+
+      - name: Upload old binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: old-aleph-node
+          path: target/release/aleph-node
+          if-no-files-found: error
+          retention-days: 7
+
+      - name: Cleanup cache
+        uses: ./.github/actions/post-cache
+
+  build-new-runtime-and-try_runtime:
+    name: Build new runtime and try_runtime tool
+    needs: [ check-runtime-change ]
+    if: ${{ needs.check-runtime-change.outputs.runtime-updated != 0 }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+
+      - name: Install WASM target
+        run: rustup target add wasm32-unknown-unknown
+
+      - name: Restore cache
+        uses: ./.github/actions/restore-cache
+        with:
+          target-key: try-runtime
+          cargo-key: new
+
+      - name: Build
+        run: cargo build --release -p aleph-node --features "try-runtime short_session"
+
+      - name: Upload new runtime
+        uses: actions/upload-artifact@v3
+        with:
+          name: aleph-runtime-try_runtime
+          path: target/release/wbuild/aleph-runtime/aleph_runtime.compact.wasm
+          if-no-files-found: error
+          retention-days: 7
+
+      - name: Upload try-runtime
+        uses: actions/upload-artifact@v3
+        with:
+          name: try_runtime
+          path: target/release/aleph-node
+          if-no-files-found: error
+          retention-days: 7
+
+      - name: Cleanup cache
+        uses: ./.github/actions/post-cache
+
+  test-runtime-update:
+    name: Test runtime update with try-runtime tool
+    runs-on: ubuntu-latest
+    needs: [ build-old-node, build-new-runtime-and-try_runtime ]
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Download all artifacts
+        uses: actions/download-artifact@v3
+        with:
+          path: local-tests/
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+
+      - name: Test update
+        timeout-minutes: 20
+        env:
+          OLD_BINARY: old-aleph-node/aleph-node
```

Comment:
If we go with the second approach mentioned above, we just need to build one binary more (with `--try-runtime` feature), instead of building two.

Reply:
done